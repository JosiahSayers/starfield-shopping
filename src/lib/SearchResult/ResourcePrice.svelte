<script lang="ts">
  import credit from "../../assets/credit.png";

  export let price: number;
</script>

<span class="badge badge-neutral m-1"
  ><img src={credit} alt="starfield credit logo" class="h-4" />{price}</span
>
