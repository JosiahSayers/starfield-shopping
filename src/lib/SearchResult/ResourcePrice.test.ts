import { describe, afterEach, it, expect } from 'vitest';
import { cleanup, render } from '@testing-library/svelte'
import ResourcePrice from './ResourcePrice.svelte'

describe('ResourcePrice', () => {
  afterEach(() => cleanup())

  it('mounts', () => {
    const { container } = render(ResourcePrice, { price: 0 });
    expect(container).toBeTruthy();
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders the price', () => {
    const { container } = render(ResourcePrice, { price: 42 });
    expect(container).toBeTruthy();
    expect(container.querySelector('.badge')?.textContent).toBe('42');
  });
});
