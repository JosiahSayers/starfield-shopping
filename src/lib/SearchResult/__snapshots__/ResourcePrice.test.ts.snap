// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResourcePrice > mounts 1`] = `"<div><span class=\\"badge badge-neutral m-1\\"><img src=\\"/src/assets/credit.png\\" alt=\\"starfield credit logo\\" class=\\"h-4\\">0</span></div>"`;
